tensorflow_datasets/core/visualization.py

```diff
@@ -85,46 +85,82 @@ def show_examples(ds_info, ds, rows=3, cols=3, plot_scale=3., image_key=None):
 
     image_key = image_keys[0]
 
-  label_keys = [
-      k for k, feature in ds_info.features.items()
-      if isinstance(feature, features_lib.ClassLabel)
-  ]
-
-  label_key = label_keys[0] if len(label_keys) == 1 else None
-  if not label_key:
-    logging.info("Was not able to auto-infer label.")
-
-  num_examples = rows * cols
-  examples = list(dataset_utils.as_numpy(ds.take(num_examples)))
+    label_keys = [
+        k for k, feature in ds_info.features.items()
+        if isinstance(feature, features_lib.ClassLabel)
+    ]
 
-  fig = plt.figure(figsize=(plot_scale*cols, plot_scale*rows))
-  fig.subplots_adjust(hspace=1/plot_scale, wspace=1/plot_scale)
-  for i, ex in enumerate(examples):
-    if not isinstance(ex, dict):
-      raise ValueError(
-          "tfds.show_examples requires examples as `dict`, with the same "
-          "structure as `ds_info.features`. It is currently not compatible "
-          "with `as_supervised=True`. Received: {}".format(type(ex)))
-    ax = fig.add_subplot(rows, cols, i+1)
-
-    # Plot the image
-    image = ex[image_key]
-    if len(image.shape) != 3:
-      raise ValueError(
-          "Image dimension should be 3. tfds.show_examples does not support "
-          "batched examples or video.")
-    _, _, c = image.shape
-    if c == 1:
-      image = image.reshape(image.shape[:2])
-    ax.imshow(image, cmap="gray")
-    ax.grid(False)
-    plt.xticks([], [])
-    plt.yticks([], [])
-
-    # Plot the label
-    if label_key:
-      label = ex[label_key]
-      label_str = ds_info.features[label_key].int2str(label)
-      plt.xlabel("{} ({})".format(label_str, label))
-  plt.show()
-  return fig
+    label_key = label_keys[0] if len(label_keys) == 1 else None
+    if not label_key:
+      logging.info("Was not able to auto-infer label.")
+
+    num_examples = rows * cols
```

Comment:
Add a check here `if image_key` so that below code runs only if data is image dataset

Reply:
Refer to code line 78 - This is all under if image keys: indicating that the code runs only when the image keys have been successfully generated.